Merge Mongodb & Hibernate5

jdaugherty opened the pull request:
This PR merges the mongodb & hibernate5 projects.

jamesfredley committed:
use Groovy 4.0.24-SNAPSHOT

jamesfredley committed:
move grails-bom to resolve dependency versions for all subprojects

jamesfredley committed:
use h2 version from grails-bom

jamesfredley committed:
restore domain inheritance tests

jamesfredley committed:
fix failing tests by addressing h2 reserved words

jamesfredley committed:
Merge pull request #910 from grails/restore-domain-inheritance

Use grails-bom, grails-bom h2 version, fix h2 tests & restore domain inheritance tests

jamesfredley committed:
Merge branch '9.0.x' into renovate/spring-boot

jamesfredley committed:
Merge pull request #908 from grails/renovate/junit5-monorepo

fix(deps): update junit5 monorepo

jamesfredley committed:
compileJava options release one line

codeconsole committed:
Merge pull request #909 from grails/renovate/spring-boot

fix(deps): update spring boot to v3.3.5

jdaugherty committed:
Update for H2 reserved keywords, Add .sdkmanrc, Update to ContainerGebSpec, Remove @Ignore on tests fixed by classpath scanning fixes

jdaugherty committed:
Add workaround for GROOVY-11512

jdaugherty committed:
Add workaround for json $target fields.

jdaugherty committed:
fix #1155 - skipValidate logic in GormValidateable causes missed validations and is inconsistent

jdaugherty committed:
Merge pull request #912 from jdaugherty/9.0.x

jdaugherty committed:
Merge pull request #913 from jdaugherty/feature/9.0.x/1155-skip-validate

fix #1155 - skipValidate logic in GormValidateable causes missed validations and is inconsistent

jamesfredley committed:
Update error exceptions from javax to jakarta

jamesfredley committed:
Merge pull request #914 from grails/javax-jakarta-error-exception

Update error exceptions from javax to jakarta

codeconsole committed:
Rename hibernate5Version -> hibernateVersion

codeconsole committed:
Merge pull request #915 from codeconsole/9.0.x-rename-version

Rename hibernate5Version -> hibernateVersion

codeconsole committed:
clean up spring dependencies

codeconsole committed:
Remove no longer needed configuraitons.all closures

codeconsole committed:
Merge pull request #918 from codeconsole/9.0.x-bom-cleanup

Clean up Spring dependencies

codeconsole committed:
picocliVersion not in bom

jamesfredley committed:
restore examples grails data services tests

jamesfredley committed:
Merge pull request #921 from grails/restore-example-grails-data-services-test

restore examples grails data services tests

codeconsole committed:
.sdkmanrc

codeconsole committed:
jdk 17

codeconsole committed:
Merge pull request #922 from codeconsole/9.0.x

Merge 8.1.x

matrei committed:
fix: Make Spring Boot Plugin autoconfiguration work (#924)

renovate[bot] committed:
fix(deps): update groovy monorepo to v4.0.24

jdaugherty committed:
Merge pull request #925 from grails/renovate/groovy-monorepo

jdaugherty committed:
Remove snapshot repo since groovy 4.0.24 is released

jdaugherty committed:
Merge pull request #926 from jdaugherty/9.0.x

Remove snapshot repo since groovy 4.0.24 is released

jdaugherty committed:
Add back snapshot support for groovy

jdaugherty committed:
Merge pull request #927 from jdaugherty/9.0.x

Add back snapshot support for groovy

renovate[bot] committed:
fix(deps): update dependency org.apache.tomcat:tomcat-jdbc to v10.1.33

codeconsole committed:
Gradle 8.11

codeconsole committed:
Merge pull request #930 from codeconsole/9.0.x

Gradle 8.11

codeconsole committed:
Merge pull request #928 from grails/renovate/tomcatversion

fix(deps): update dependency org.apache.tomcat:tomcat-jdbc to v10.1.33

renovate[bot] committed:
chore(deps): update plugin com.gradle.enterprise to v3.18.2

renovate[bot] committed:
chore(deps): update dependency gradle to v8.11.1

matrei committed:
Use versions from `grails-bom` in example apps (#920)

* fix(deps): Use versions from `grails-bom` in example apps

And remove unused versions from `gradle.properties`

* build: Remove `webdriver-binaries-gradle-plugin`

As `ContainerGebSpec` is used now, `webdriver-binaries-gradle-plugin` can be removed.

* chore: Remove outdated comment

* chore: Cleanup `examples-grails3-database-per-tenant`

- `AnotherBookService`: Make `save()` method use passed book title
- `Application`: Remove unused imports and old comment
- `application.yml`: Format properly with 2 spaces and group config settings better
- `logback.xml`: Correct faulty pattern
- `build.gradle`: Use correct dependencies and narrow the scopes. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

* chore: Cleanup `examples-grails3-hibernate5`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.
- `BookControllerUnitSpec`: Remove unused imports and bean definition of deprecated `MimeTypesGrailsPlugin`.
- `UrlMappings`: Move to package so it is pickup up.

* chore: Cleanup `examples-grails3-multiple-datasources`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Use `astTransformation` configuration for dependencies that needs to be both `compileOnly` and `integrationTestCompileOnly`.

* chore: Cleanup `examples-grails3-partitioned-multi-tenancy`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `AnotherBookService`: Make `save()` method use passed book title
- `PartionedMultiTenancySpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

* chore: Cleanup `examples-grails3-schema-per-tenant`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `AnotherBookService`: Make `save()` method use passed book title
- `SchemaPerTenantSpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

* chore: Cleanup `examples-grails-hibernate-groovy-proxy`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `Customer`: Add no-args constructor needed for proxying
- `ProxySpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Change version variable name from `hibernateGroovyProxy` to `yakworksHibernateGroovyProxyVersion`.

* chore: Cleanup `examples-spring-boot-hibernate5`

- `application.yml`: Format properly with 2 spaces remove unused config.
- `Application`: Exclude `HibernateJpaAutoConfiguration` to avoid duplicate transaction managers: java.lang.ClassCastException: class org.springframework.orm.jpa.EntityManagerHolder cannot be cast to class org.springframework.orm.hibernate5.SessionHolder (org.springframework.orm.jpa.EntityManagerHolder and org.springframework.orm.hibernate5.SessionHolder are in unnamed module of loader 'app')
- `BookController`: Add `@ReadOnly` to `books()` action to get a session for the `Book.list()` method.
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Add version variable `springBootGradlePluginVersion`.

* test: Extract test for issue 450 from `examples-grails-data-service`

The fix for grails/gorm-hibernate5#450 did not really fit in `examples-grails-data-service`.

This commit adds a new example project, that shows the problem and that it now works (taken from https://github.com/grails-core-issues-forks/multitenantcomposite).

* chore: Cleanup `examples-grails-data-service`

- `MultitenantBook.groovy`: Remove as extracted to examples-issue-450
- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `ClassUsingAService`: Remove dependency on `grails-spring-security-rest`. No need to depend on outside plugins for this test. New test `ServiceInjectionSpec` exercising issue 202.
- `TestService`: Remove unused `transactionService` property
- `logback.xml`: Correct faulty pattern

* chore: Cleanup `examples-standalone-hibernate5`

Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

---------

Co-authored-by: Scott Murphy <scott@alwaysvip.com>

codeconsole committed:
Merge branch '9.0.x' into renovate/gradle-8.x

codeconsole committed:
Merge pull request #935 from grails/renovate/gradle-8.x

chore(deps): update dependency gradle to v8.11.1

codeconsole committed:
Merge branch '9.0.x' into renovate/com.gradle.enterprise-3.x

codeconsole committed:
Merge pull request #931 from grails/renovate/com.gradle.enterprise-3.x

chore(deps): update plugin com.gradle.enterprise to v3.18.2

codeconsole committed:
Remove versioning from example folder names

codeconsole committed:
remove version number from folder name

codeconsole committed:
Merge pull request #936 from codeconsole/9.0.x-directory-rename

Remove versioning from example folder names

jdaugherty committed:
Updating for grails-publish plugin change

jdaugherty committed:
Merge pull request #388 from jdaugherty/6.0.x

Updating for grails-publish plugin change

codeconsole committed:
17.0.13-librca

codeconsole committed:
Gradle 8.11.1

codeconsole committed:
add eof

jamesfredley committed:
Revert "fix: Remove dependency on grails-shell (#350)"

This reverts commit 488dc2756bf3df572a478c99cd8b27946a245cf9.

jamesfredley committed:
update to Grails 6.2.1 and Groovy 3.0.23

jamesfredley committed:
Merge pull request #389 from grails/restore-grails-shell-commands-scripts

Restore grails-shell dependent commands and scripts

jamesfredley committed:
Merge branch '5.0.x' into merge-5.0.x-into-6.0.x-12-11-2024

# Conflicts:
#	build.gradle
#	gradle.properties

jamesfredley committed:
update command to pass parameter of type List<T> instead of new ArrayList()

jamesfredley committed:
Merge pull request #390 from grails/merge-5.0.x-into-6.0.x-12-11-2024

Merge 5.0.x into 6.0.x

jamesfredley committed:
add javaparser-core dependency

jamesfredley committed:
Merge pull request #391 from grails/fix-groovydocs-task

add javaparser-core dependency to fix groovydoc task

jamesfredley committed:
Update README.md for 5.0.0 release

jamesfredley committed:
Merge pull request #392 from grails/pre-release-5.0.0

Update README.md for 5.0.0 release

puneetbehl committed:
[skip ci] Release v5.0.0

puneetbehl committed:
chore: Bump version to 5.0.1-SNAPSHOT

jamesfredley committed:
Merge branch '5.0.x' into merge-5.0.x-into-6.0.x-12-18-2024

# Conflicts:
#	build.gradle
#	gradle.properties

jamesfredley committed:
Update docs for 6.0.x for Grails 7

jamesfredley committed:
Merge pull request #393 from grails/merge-5.0.x-into-6.0.x-12-18-2024

Merge 5.0.x into 6.0.x

jamesfredley committed:
Use grails-bom, ContainerGebSpec, cleanup dependencies and fix tests

jamesfredley committed:
remove unused ext variables

jamesfredley committed:
Use versions from grails-bom -> spring-boot-dependencies

jamesfredley committed:
update dependencies

jamesfredley committed:
remove unused property usesGeb

jamesfredley committed:
remove unneeded dependencies

jamesfredley committed:
Merge pull request #774 from grails/more-grails-7-updates

Use grails-bom, ContainerGebSpec, cleanup dependencies and fix tests

jamesfredley committed:
Merge branch '9.0.x' into renovate/supercharge-mongodb-github-action-1.x

jamesfredley committed:
Merge pull request #760 from grails/renovate/supercharge-mongodb-github-action-1.x

chore(deps): update supercharge/mongodb-github-action action to v1.11.0

renovate[bot] committed:
chore(deps): update dependency gradle to v8.12

jamesfredley committed:
use more versions from grails-bom

jamesfredley committed:
Merge pull request #945 from grails/use-more-grails-bom-version

use more versions from grails-bom

jdaugherty committed:
Move views-json-templates from grails-views to gorm-mongodb

jdaugherty committed:
Merge pull request #783 from jdaugherty/9.0.x

Move views-json-templates from grails-views to gorm-mongodb

jdaugherty committed:
Add back hibernateValidatorVersion

jdaugherty committed:
Fix bom import, use liberica, use grails github actions, support the pre release workflow

# Conflicts:
#	docs/build.gradle
#	gradle.properties
#	grails-plugin/build.gradle

jdaugherty committed:
Merge pull request #940 from jdaugherty/9.0.x

Pre-release for M1

grails-build committed:
[skip ci] Release v9.0.0-M2

grails-build committed:
chore: Bump version to 9.0.0-SNAPSHOT

jdaugherty committed:
Fixes found after pre release for mongodb

jdaugherty committed:
Fix bom import, use liberica, use grails github actions, support the pre release workflow

jdaugherty committed:
Merge pull request #784 from jdaugherty/9.0.x

Fix bom import, use liberica, use grails github actions, support the pre release workflow

grails-build committed:
chore: Bump version to 9.0.0-SNAPSHOT

jdaugherty committed:
Merge pull request #947 from jdaugherty/9.0.x

Fixes found after pre release for mongodb

jdaugherty committed:
[skip ci] pre release workflow support

grails-build committed:
[skip ci] Release v6.0.0-M1

grails-build committed:
chore: Bump version to 6.0.0-SNAPSHOT

jamesfredley committed:
use grails-datastore-gorm-tck version from grails-bom

jamesfredley committed:
use grails-datastore-gorm-tck version from grails-bom

jamesfredley committed:
remove unused version

jamesfredley committed:
Merge pull request #949 from grails/use-tck-from-grails-bom

use grails-datastore-gorm-tck version from grails-bom

jamesfredley committed:
Merge pull request #786 from grails/use-tck-from-grails-bom

use grails-datastore-gorm-tck version from grails-bom

codeconsole committed:
Merge pull request #780 from grails/renovate/gradle-8.x

chore(deps): update dependency gradle to v8.12

codeconsole committed:
.sdkmanrc

codeconsole committed:
Use mongodbDriver 5.3.0

codeconsole committed:
Merge pull request #788 from codeconsole/9.0.x-5.3.0

Use mongodbDriver 5.3.0

matrei committed:
build: add system property for TCK test control (#790)

Introduce a new system property, `mongodb.gorm.suite`, to manage the enabling or disabling of specific tests within the TCK.

jamesfredley committed:
exclude hibernate-core dependency

jamesfredley committed:
add jboss-transaction-api dependency

jamesfredley committed:
remove doc change from Grails 3

matrei committed:
build: remove module dependency on `hibernate-ehcache`

Moved `hibernate-ehcache` to a runtime dependency where needed for tests, making it optional and reducing unnecessary module coupling.

As a result, applications using `hibernate-ehcache` will now need to explicitly add it as a dependency.

matrei committed:
build: explicitly exclude `hibernate-core` from `hibernate-ehcache`

Even if it does not cause issues in this project, we explicitly exclude
the `javax` variant of `hibernate-core` from `hibernate-ehcache` to
document best practices and prevent potential conflicts in other setups.

matrei committed:
build: add `jboss-transaction-api` for `hibernate-ehcache`

Excluding the `javax` variant from `hibernate-ehcache` requires explicitly
including `jboss-transaction-api`.

matrei committed:
docs: update GORM in Spring Boot section

- Revise the instructions for using GORM in a Spring Boot application.
- Simplify the `application.yml` file in the Spring Boot test project.

matrei committed:
build: fix wrong dependency configuration

matrei committed:
fix: enable correct Spring auto-configuration in `boot-plugin`

The Spring auto-configuration was incorrectly set in the `boot-plugin`, causing misconfiguration issues. This commit ensures the correct auto-configuration is applied.

matrei committed:
ci: fix typo in `renovate.json` for `grails-datastore-async`

Corrected a typo in the package name grails-datastore-async in renovate.json to ensure proper dependency updates.

matrei committed:
ci: update GitHub workflows

Update the GitHub workflows to the latest Grails repository standard and best practices.

matrei committed:
test: fix `MongoDbGormAutoConfigurationSpec` failure due to misconfiguration

Fixed a Spring misconfiguration that caused MongoDbGormAutoConfigurationSpec to fail. This commit ensures the test runs correctly.

matrei committed:
build: extract common build logic into `*-config.gradle` files

Moved shared build configuration logic into `*-config.gradle` files, enabling a more compositional approach to build setup.

matrei committed:
test: update `BsonPersistentEntityCodecSpec` for updated API

Refactored `BsonPersistentEntityCodecSpec` to align with the updated API. The constructor `JsonWriterSettings(JsonMode)`` no longer exists, requiring adjustments to the test.

matrei committed:
test: refactor `BookSpec` to not use try/catch in setup

matrei committed:
test: remove Micronaut HttpClient dependency from `GlobalTemplateSpe`

Refactored GlobalTemplateSpec to eliminate the use of Micronaut HttpClient, removing the project's dependency on it.

matrei committed:
test: re-enable two passing tests in `MongoDbDataStoreSpringInitializerSpec`

Removed the `@Ignore` annotation from two tests in `MongoDbDataStoreSpringInitializerSpec` since they are now passing, allowing them to run as part of the test suite.

matrei committed:
build: update develocity build scan and remote cache

matrei committed:
test: clean up config in test applications

matrei committed:
refactor: remove `grails-app` from Grails MongoDB Plugin

Remove unused `grails-app` directory.

The `application.yml` setting `grails.gorm.failOnError: false` seems to be a
left-over from copy-pasting the `gorm-hibernate` plugin.

matrei committed:
fix: update `MongodbGrailsPlugin` for Grails 7

matrei committed:
chore: cleanup unused imports

matrei committed:
test: replace `@Ignore` with `@PendingFeature` in `LocalMongoUnitSpec`

Update a test in `LocalMongoUnitSpec` to use `@PendingFeature` instead of `@Ignore`,
allowing it to run while marking it as an expected failure until the issue is resolved.

matrei committed:
test: replace deprecated `Package.getPackage()` usage in `MongoSpec`

Refactor `MongoSpec` to remove the deprecated `Package.getPackage()` call,
ensuring compatibility with newer Java versions.

matrei committed:
test: cleanup `UrlMappings` in `examples/gson-templates`

It looks like this file has been copy/pasted from another project.

matrei committed:
build: refactor build configuration and update dependencies

- Adopt a composition-based build structure.
- Integrate the Grails Publish Plugin.
- Update dependencies and refine dependency scopes.
- Add support for the `excludeUnusedTransDeps` flag to allow building and testing with unused transitive dependencies excluded.

matrei committed:
ci: update mongodb settings

- Update 7.0.14 to 7.0 which uses the latest 7.0.x version
- Add mongodb server to groovy-joint-workflow.yml

matrei committed:
ci: update mongodb versions

- Switch to only use major version.
- Add version 8.

matrei committed:
chore: remove author and developer info from `MongodbGrailsPlugin`

Author and developer metadata have been removed from MongodbGrailsPlugin, as this information is not necessary for core plugins.

matrei committed:
build: fix maven pom info for `views-json-templates`

matrei committed:
chore: remove unused version property `grailsViewsVersion`

matrei committed:
ci: use v3 tag for `grails/github-actions`

matrei committed:
ci: use v3 tag for `grails/github-pages-deploy-action`

matrei committed:
Merge pull request #793 from grails/matrei/project-overhaul

Multiple Project updates

jdaugherty committed:
Doc fixes from data-mapping merge attempt

jdaugherty committed:
Merge pull request #794 from jdaugherty/9.0.x

Doc fixes from data-mapping merge attempt

matrei committed:
ci: fix snapshot publishing by adding auth details (#795)

The snapshot publishing step was failing due to missing authentication details. This commit adds the necessary credentials to ensure successful publishing.

jamesfredley committed:
back to grails 7.0.0-SNAPSHOT

jamesfredley committed:
Merge pull request #398 from grails/back-to-grails-7.0.0-SNAPSHOT

back to grails 7.0.0-SNAPSHOT

gsartori committed:
BootStrap init closure servletContext cleanup

gsartori committed:
BootStrap init closure servletContext cleanup

gsartori committed:
BootStrap init closure servletContext cleanup

renovate[bot] committed:
Update plugin com.gradle.enterprise to v3.19.2

jdaugherty committed:
Merge pull request #957 from gsartori/9.0.x

BootStrap init closure servletContext cleanup

jdaugherty committed:
Merge pull request #797 from gsartori/9.0.x

BootStrap init closure servletContext cleanup

jamesfredley committed:
remove gradle, github workflows, .gitignore and license

(cherry picked from commit cffd761f5d3c4b447fbb026f9af4091539199803)
(cherry picked from commit 1611aa1d7311bb04faf34a500c340410ba6fc99f)

jamesfredley committed:
update build

(cherry picked from commit 12aa4b9e50943b7429856f1bab4980b0fa5614fe)
(cherry picked from commit 55334a55564ade8f1ae38b70903091f11132285d)

jamesfredley committed:
asciidoc consolidation into /docs

(cherry picked from commit 788d2b0f63de50eeab81ee4b8d375ddb2e78a9f6)
(cherry picked from commit 769cf1906e5993397e30c53ef868657c879d2055)

jamesfredley committed:
Update Grails 7 version to 9.0.x and point to new documentation location

(cherry picked from commit 3ce0df9750832bd0dc094f66d1e50cd2a1600e89)
(cherry picked from commit 4c0fc6b4fa1cc495d1a77bb8cfd65a000eb211e4)

jamesfredley committed:
Merge pull request #397 from MichaelMorett/feature/fix-getting-started

docs: mentioned that the placement of sourceSets affects Grails 4 and above
(cherry picked from commit 4cc66c98068f17dea83e58839fb246230949f305)

jamesfredley committed:
Merge pull request #399 from grails/5.0.x

Merge pull request #397 from MichaelMorett/feature/fix-getting-started

jamesfredley committed:
Merge remote-tracking branch 'grails-database-migration/6.0.x' into merge-database-migration-plugin

jamesfredley committed:
Merge pull request #955 from grails/exclude-hibernate-core

Exclude hibernate-ehcache

jamesfredley committed:
Update example projects to set hibernate-ehcache dependency

While excluding hibernate-core (javax) and including jboss-transaction-api

jamesfredley committed:
document hibernate-ehcache breaking change in documentation

jdaugherty committed:
Update docs to be correctly nested and fix attribute issues

jamesfredley committed:
Remove ehcache configuration from gson-templates since it does not use hibernate

jamesfredley committed:
Merge pull request #961 from grails/hibernate-ehcache-breaking-change

document hibernate-ehcache breaking change in documentation

jamesfredley committed:
Merge branch '9.0.x' into merge-database-migration-plugin

# Conflicts:
#	gradle.properties

jamesfredley committed:
Merge pull request #800 from grails/hibernate-ehcache-breaking-change

Update example projects to set hibernate-ehcache dependency

jamesfredley committed:
Merge pull request #959 from grails/merge-database-migration-plugin

Merge database migration plugin into gorm-hibernate5

acanby committed:
Allow loading classpath resources for sqlFile changesets

nobeans committed:
Remove "dataSource_" from bean name as dataSourceName of GroovyChange when using additional dataSources

jdaugherty committed:
Removing documented command that was removed in commit 23b3ca66

jdaugherty committed:
Nullsafe

jdaugherty committed:
Update description of groovy change

jdaugherty committed:
Merge pull request #966 from grails/issue965

Remove "dataSource_" from bean name as dataSourceName of GroovyChange when using additional dataSources

jdaugherty committed:
Handle loading migrations from embedded jars (plugins)

zyro23 committed:
fix grails/gorm-hibernate5#973 (9.0.x)

jdaugherty committed:
Merge pull request #985 from zyro23/issue-973

fix grails/gorm-hibernate5#973 for grails-7 (9.0.x)

jdaugherty committed:
Exclude liquibase-commercial from being exported

jdaugherty committed:
Merge pull request #971 from grails/groovyChangeDescription

Update description of groovy change

jdaugherty committed:
Merge pull request #969 from grails/issue968

Removing documented command that was removed in commit 23b3ca66

jdaugherty committed:
Merge pull request #986 from grails/excludeLiquibaseCommerical

Exclude liquibase-commercial from being exported

jdaugherty committed:
Merge pull request #981 from grails/embeddedJarHandling

Handle loading migrations from embedded jars (plugins)

jdaugherty committed:
Merge pull request #964 from grails/issue963

Allow loading classpath resources for sqlFile changesets

renovate[bot] committed:
Update dependency gradle to v8.13

jamesfredley committed:
remove groovyVersion from gradle.properties

jamesfredley committed:
Use snapshot versions

jamesfredley committed:
use grails-bom version in buildscript{}

jamesfredley committed:
Merge pull request #989 from grails/remove-groovyVersion

remove groovyVersion from gradle.properties

renovate[bot] committed:
Update dependency org.asciidoctor:asciidoctor-gradle-jvm to v4.0.4

jamesfredley committed:
Merge branch '9.0.x' into renovate/gradle-8.x

jamesfredley committed:
Merge branch '9.0.x' into renovate/com.gradle.enterprise-3.x

jamesfredley committed:
Merge pull request #916 from grails/renovate/asciidoctorgradleversion

Update dependency org.asciidoctor:asciidoctor-gradle-jvm to v4.0.4

jamesfredley committed:
Merge pull request #939 from grails/renovate/com.gradle.enterprise-3.x

Update plugin com.gradle.enterprise to v3.19.2

renovate[bot] committed:
Update plugin com.gradle.common-custom-user-data-gradle-plugin to v2.1

jamesfredley committed:
Merge pull request #944 from grails/renovate/gradle-8.x

Update dependency gradle to v8.13

jamesfredley committed:
Merge pull request #953 from grails/renovate/com.gradle.common-custom-user-data-gradle-plugin-2.x

Update plugin com.gradle.common-custom-user-data-gradle-plugin to v2.1

jamesfredley committed:
use spring-boot-gradle-plugin version from grails-bom

jamesfredley committed:
add repo.grails.org

jamesfredley committed:
Merge pull request #990 from grails/use-more-version-grails-bom

use spring-boot-gradle-plugin version from grails-bom

jdaugherty committed:
[skip ci] grails 7.0.0-M3 release

jdaugherty committed:
[skip ci] copy the staged release workflow from the data-mapping project

jdaugherty committed:
[skip ci] fix publishing

grails-build committed:
[skip ci] Release v9.0.0-M3

jdaugherty committed:
[skip ci] grails 7.0.0-M3

jdaugherty committed:
[skip ci] switch to datastore version 9.0.0-M3

grails-build committed:
[skip ci] Release v9.0.0-M3

jdaugherty committed:
[skip ci] add missing github pre release repo

jdaugherty committed:
[skip ci] add missing github pre release repo

grails-build committed:
chore: Bump version to 9.0.0-SNAPSHOT

jdaugherty committed:
[skip ci] add staged repo

grails-build committed:
chore: Bump version to 9.0.0-SNAPSHOT

jamesfredley committed:
Merge pull request #988 from dertalai/7.2.x

Fix java.lang.NullPointerException when incorrect schema in database.

(cherry picked from commit eff81e52091ee975706eeaad39d6263e66687b8c)

jdaugherty committed:
Merge pull request #991 from grails/988-merge-to-9.0.x

Merge pull request #988 from dertalai/7.2.x

matrei committed:
chore: major project cleanup

Goal of cleanup was to use the `org.grails.grails-publish` plugin
and use a more composition based build approach.

matrei committed:
ci: fix problems

matrei committed:
build: add the `slf4jPreventExclusion` property

jamesfredley committed:
use GrailsStringUtils.trimToNull() instead of apache commons version

https://github.com/grails/grails-core/pull/13807

jamesfredley committed:
Merge pull request #993 from grails/use-GrailsStringUtils-trimToNull

use GrailsStringUtils.trimToNull() instead of apache commons version

matrei committed:
build: pr feedback

matrei committed:
build: updated pr feedback

matrei committed:
Merge pull request #992 from grails/matrei/project-cleanup

- Use ´org.grails.grails-publish` for publishing
- Use composition based build

jdaugherty committed:
Imported grails-data-hibernate5

jdaugherty committed:
Integrate grails-data-hibernate5

jdaugherty committed:
Imported mongodb

jdaugherty committed:
Integrate Mongodb

jdaugherty committed:
Update for spring security plugin move & remove mavenLocal()

jdaugherty committed:
Split hibernate5, mongodb, and datastore tests

jamesfredley committed:
Merge branch '9.0.x' into newMongoMerge

jamesfredley commented:
I think we should rename `hibernate5` and `mongodb `to start with `grails-data-`, like `neo4j` and `graphql `that were brought over.

jamesfredley reviewed (approved):
These look good:

https://github.com/grails/grails-data-mapping/pull/1933/commits/834941e73bfbf47cbf84e5933e9f9f9d8af7fb8e

https://github.com/grails/grails-data-mapping/pull/1933/commits/88a565af1834455c1694d94e4bdb45357a688b8e

https://github.com/grails/grails-data-mapping/pull/1933/commits/ab43a143fad12a16d8f52bb43bfeeb2c9f46510b

https://github.com/grails/grails-data-mapping/pull/1933/commits/480fe9aeb1a486c739eb23a3adeb72c359802f0b

jamesfredley committed:
Add missing mongodb-version's

codeconsole commented:
Why?  What is the win here?  Hibernate 5 is dead. You are literally merging a project with [91 open issues](https://github.com/grails/grails-data-hibernate5/issues) that is an obsolete implementation with no future into a project that has[ 218 open issues](https://github.com/grails/grails-data-mapping/issues).  Hibernate 5.6 has been in [limited support mode](https://hibernate.org/orm/releases/5.6/) for quite some time. 

### There hasn't been a single release of [Hibernate 5.x](https://mvnrepository.com/artifact/org.hibernate/hibernate-core/5.6.15.Final)  in OVER 2 years

All this does is further obfuscate the codebase and couple the data mapping repo to a dead implementation where significant work was done to separate it in the first place. 

91 + 218 + 47 open issues and 5,418 commits merged into a repo with only 9,788 commits thus adding 55% to the commit history.

jamesfredley commented:
They should have always been consolidated so they are updated, built and released together.  With a volunteer only team this is the only viable path forward.  The internet is blazing fast and storage is cheap, so the size and number of commits are far down the list of priorities and the focus is on time to maintain and release the projects.  The amount of time sunk into keep them separate over the last 9 months is staggering and should have been used for https://github.com/orgs/grails/projects/3 instead.

https://github.com/orgs/grails/discussions/13521#discussioncomment-12451797 is the cold hard facts about the current hibernate 5, 6 & 7 situation.

jamesfredley committed:
add mongodb-github-action to groovy-joint-workflow

jamesfredley committed:
force dependency refresh

codeconsole commented:
@jamesfredley If any time was truly invested keeping them separate, all that time is now wasted by merging them together. 

My point in adding 5,418 commits had absolutely nothing to do with repo size and the amount of time it is going to take to clone the repo, but you actually gave another reason why not to do it.  If this project has any future, this O(1) commit will have an O(N) effect on build times indefinitely.   

My point was centered on polluting the commit history by almost 60% for commits irrelevant to the underlying project.  Commit history serves purpose and should be easy to navigate and not require additional steps.  Furthermore, adding 63% more open outstanding issues to the project doesn't help supporting the project by any means.

There is no reason why they should have to be released together. They should have the ability to be released independently.  Mongo is an active implementation that is under active development.  It should have a faster release cycle to accommodate dependency updates which happen on a regular basis.  As I have previously stated, hibernate 5.x hasn't had a single release in over 2 years. It's dependencies are stale and obsolete. 

If you could please be more specific on why exactly they need to be built and released together and how their current state makes things complicated. Perhaps the better approach would be to addressing the underlying problem with a scalpel rather than a nuclear bomb.

jamesfredley committed:
fix gradle.yml

jamesfredley commented:
@codeconsole The quorum from the main contributors was to consolidate, so this is moving forward.   

It will likely make your future contributions more efficient across the full process to release.

jamesfredley committed:
force refresh dependencies

jamesfredley committed:
adjust matrix strategy and add mongodb-github-action for datastore build

jamesfredley committed:
try again

jdaugherty committed:
Code review feedback:
- rename directory to include grails-data
- distinguish migraiton plugin project as hibernate5 (leave artifact as is for now)
- workaround for asset pipeline #177

jamesfredley committed:
Revert "force refresh dependencies"

This reverts commit b9d15e135a6428aa4674cb72c3bf624ae72a320e.

jamesfredley committed:
Revert "force dependency refresh"

This reverts commit 739b47b2caafd7ae792690a1ee1df9470a140e03.

codeconsole commented:
@jamesfredley so there is no ticket or discussion on why exactly they need to be built and released together?  All I am wanting to understand is what is the gain in merging these 2 repos into data-mapping and how that can't be accomplished in another way while keeping them apart.   I think this merge will be a major mistake.

jdaugherty commented:
@codeconsole we have talked about this for months at this point.   Every time you've been the only person to disagree and everyone else , especially those that perform the builds, have said this is for the best.

We have agreed to it multiple times previously, and then backed off because it wasn't technically possible.  With the project restructures, we realized it was possible given the previous approval, we are moving forward.  

For the short term future, this project will be maintained.  We can't easily do that when it's separate - especially given the limited volunteer time.  We are about to repackage as part of the move.  This is being moved and built to be continued supported until we have a hibernate 6 implementation.

codeconsole commented:
@jdaugherty yes, I was the 1 person to disagree with 3 other people during a 3 minute discussion because I think it is an incorrect approach to solve an underlying problem that still has yet to be addressed: the unnecessary coupling of dependencies and the bom.  This all could have been cleaned up and resolved in a week. I have been strongly against these consolidations and that is main reason why I haven't been contributing.  

Rather than focus on compatibility and finishing the Spring Boot 2 > 3 upgrade, we have lost 3 months doing project consolidations without even being able to keep up to date with patch/bug releases of the underlying Spring Boot/Spring Security/Spring Framework.  M2 was released with outdated patch versioning that came out 2 weeks prior.  

I still don't understand how merging in 3,299 unrelated commits and  91 open issues from a project based on dependencies that haven't been updated in 2 years helps anything.  I think decisions as significant as this should warrant more time than 3 minutes discussing their rationality and purpose. At the very least they should have a clear definition on what it is exactly they are accomplishing other than eliminating a build file. 

Is it really that difficult to release gorm hibernate 5 independently?  What exactly is this accomplishing?

jdaugherty commented:
The meeting minutes show this was discussed for multiple weeks with multiple people.  Please do not represent this as an immediate decision or one off.  You were the only person who disagreed.  Every time we discussed it, it was approved.  

I get that you do not use hibernate 5, but many people do.  The project consolidation occurred to facilitate development.  Having separate projects has caused a lot of pain.  We have enumerated that pain and even offered to show you it.  You have ignored any pleas for help with the build.  

All of these merges have not prevented contributions.  We welcome them.  Your further help on Spring upgrades or the multiple layout issues due to the sitemesh upgrade is welcome. 

Almost every project merger has revealed major issues.  These mergers have had a huge benefit as a result.  We are volunteers and we are positioning ourselves for faster releases.  Merging the project does not affect any function of grails; it only helps us release faster.

codeconsole commented:
> We have enumerated that pain and even offered to show you it. You have ignored any pleas for help with the build.

That is **not a true statement** . Who, and when, ever offered to show me anything of the such?  In fact, the opposite.  Not once did anyone reach out to me to help with the build.  I actually offered to fix all the build/release issues, but was voted against in favor of consolidations of which, I repeat, still haven't even addressed the underlying dependency coupling issue. 

@jdaugherty So how much time would you say this was discussed in the weekly meeting? Maximum of what, collectively 5 minutes then?  During that time was there ever a thorough explanation detailing why it was needed?  no.  Is there a solid reference to any documentation anywhere on why it is needed? no.  Wouldn't an issue or a GitHub discussion be a better place for such a discussion to be done?

> I get that you do not use hibernate 5, but many people do. 

That is also **not true**. I do use hibernate in a few projects and in the other I use mongo and that has zero impact on my opinion. 

I am still looking for a basic explanation on how exactly merging hibernate 5 into grails-data-mapping speeds up anything. 
I think it is going to slow things down and make the overall project more difficult to maintain.  If I didn't strongly feel that way, I wouldn't even be wasting my time having this discussion.

How exactly is this going to speed up the release process?  What is specifically causing an issue by keeping them separate?

jamesfredley commented:
https://app.slack.com/client/T07LCCCSU/G07LE2X60?cdn_fallback=1
https://grails.slack.com/archives/G07LE2X60/p1737138429799719?thread_ts=1737138158.749779&cid=G07LE2X60
https://grails.slack.com/archives/G07LE2X60/p1737138158749779
https://grails.slack.com/archives/G07LE2X60/p1737041798675299?thread_ts=1737010496.906699&cid=G07LE2X60
https://grails.slack.com/archives/G07LE2X60/p1737030975987449?thread_ts=1736989240.503509&cid=G07LE2X60
https://grails.slack.com/archives/G07LE2X60/p1735957212262229
https://grails.slack.com/archives/G07LE2X60/p1736187333832829?thread_ts=1735957212.262229&cid=G07LE2X60
https://grails.slack.com/archives/G07LE2X60/p1735616756887189?thread_ts=1734966750.089439&cid=G07LE2X60
https://grails.slack.com/archives/G07LE2X60/p1734966750089439

https://github.com/grails/grails-core/issues/13552

https://docs.google.com/document/d/1L590r6Z_xNQ48l31Yolsir75dUgAorEnTWmW1kGqJrw/edit?tab=t.0#heading=h.yzvjivh6d3yb

https://github.com/orgs/grails/discussions/13516

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -29,15 +30,78 @@ jobs:
+        id: build
+        env:
+          GITHUB_MAVEN_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew build --continue -PonlyDatatstoreTests
```
Typo

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -7,16 +7,17 @@ on:
       matrix:
-        java: [17, 21]
+        java: [ 17, 21 ]
+        mongodb-version: [ '4', '5', '6', '7', '8' ]
```
Do we need to start mongo here?

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -29,15 +30,78 @@ jobs:
+      - name: "▶️ Start MongoDB (needed for some tests)"
+        uses: supercharge/mongodb-github-action@90004df786821b6308fb02299e5835d0dae05d0d # v1.12.0 (Use commit sha as this is a 3rd party action)
+        with:
+          mongodb-version: ${{ matrix.mongodb-version }}
```
Do we need to start mongo here?

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -7,16 +7,17 @@ on:
+  datastore-build:
+    name: "Build Datastore Project"
     runs-on: ubuntu-24.04
     permissions:
       contents: read
       packages: read
```
Can we move the permissions to the workflow level, so we don't have to repeat it on every job?

jdaugherty replied:
The permissions are different across these workflows - docs will publish to a branch, while none of the others will.

matrei replied:
Could we override with `permissions.contents: write` on the `docs` job?

jdaugherty replied:
Yes we can!  Thank you

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -29,15 +30,78 @@ jobs:
         env:
           GITHUB_MAVEN_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew build --continue
+        run: ./gradlew build --continue -PonlyMongodbTests
```
I am getting a test failure when running this locally (`example.BookControllerSpec.test find by title(BookControllerSpec.groovy)`:
(Running `./gradlew build --continue -PonlyMongodbTests --no-parallel` works.)
```console
com.mongodb.MongoBulkWriteException: Bulk write operation error on server 127.0.0.1:27017. Write errors: [BulkWriteError{index=1, code=11000, message='E11000 duplicate key error collection: test.book index: _id_ dup key: { _id: 1 }', details={}}]. 
	at app//com.mongodb.internal.connection.BulkWriteBatchCombiner.getErrorNonNullable(BulkWriteBatchCombiner.java:173)
	at app//com.mongodb.internal.connection.BulkWriteBatchCombiner.throwOnError(BulkWriteBatchCombiner.java:196)
	at app//com.mongodb.internal.connection.BulkWriteBatchCombiner.getResult(BulkWriteBatchCombiner.java:137)
	at app//com.mongodb.internal.operation.BulkWriteBatch.getResult(BulkWriteBatch.java:234)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.executeBulkWriteBatch(MixedBulkWriteOperation.java:295)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.lambda$execute$2(MixedBulkWriteOperation.java:206)
	at app//com.mongodb.internal.operation.SyncOperationHelper.lambda$withSourceAndConnection$0(SyncOperationHelper.java:131)
	at app//com.mongodb.internal.operation.SyncOperationHelper.withSuppliedResource(SyncOperationHelper.java:156)
	at app//com.mongodb.internal.operation.SyncOperationHelper.lambda$withSourceAndConnection$1(SyncOperationHelper.java:130)
	at app//com.mongodb.internal.operation.SyncOperationHelper.withSuppliedResource(SyncOperationHelper.java:156)
	at app//com.mongodb.internal.operation.SyncOperationHelper.withSourceAndConnection(SyncOperationHelper.java:129)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.lambda$execute$3(MixedBulkWriteOperation.java:191)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.lambda$decorateWriteWithRetries$0(MixedBulkWriteOperation.java:148)
	at app//com.mongodb.internal.async.function.RetryingSyncSupplier.get(RetryingSyncSupplier.java:67)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.execute(MixedBulkWriteOperation.java:210)
	at app//com.mongodb.internal.operation.MixedBulkWriteOperation.execute(MixedBulkWriteOperation.java:80)
	at app//com.mongodb.client.internal.MongoClusterImpl$OperationExecutorImpl.execute(MongoClusterImpl.java:446)
	at app//com.mongodb.client.internal.MongoCollectionImpl.executeBulkWrite(MongoCollectionImpl.java:470)
	at app//com.mongodb.client.internal.MongoCollectionImpl.bulkWrite(MongoCollectionImpl.java:450)
	at app//com.mongodb.client.internal.MongoCollectionImpl.bulkWrite(MongoCollectionImpl.java:445)
	at app//org.grails.datastore.mapping.mongo.MongoCodecSession.flush(MongoCodecSession.groovy:234)
	at app//org.grails.datastore.mapping.mongo.AbstractMongoSession.flush(AbstractMongoSession.java:73)
	at app//example.BookControllerSpec.$tt__$spock_feature_0_0(BookControllerSpec.groovy:31)
	at example.BookControllerSpec.test find by title_closure1(BookControllerSpec.groovy)
	at app//groovy.lang.Closure.call(Closure.java:433)
	at app//groovy.lang.Closure.call(Closure.java:422)
	at app//grails.gorm.transactions.GrailsTransactionTemplate$1.doInTransaction(GrailsTransactionTemplate.groovy:68)
	at app//org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:140)
	at app//grails.gorm.transactions.GrailsTransactionTemplate.executeAndRollback(GrailsTransactionTemplate.groovy:65)
	at example.BookControllerSpec.test find by title(BookControllerSpec.groovy)
```

jdaugherty replied:
I've serialized the mongo tests until we can spin up containers to run them in parallel

matrei commented on gradle/test-config.gradle:
```diff
@@ -1,8 +1,28 @@
 
 tasks.withType(Test).configureEach {
+    onlyIf {
+        if(project.hasProperty('skipDatatstoreTests')) {
```
Typo

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -97,7 +161,7 @@ jobs:
     if: github.event_name == 'push'
     name: "Trigger Build in GORM Implementations"
-    needs: [build, publish]
+    needs: [datastore-build, mongodb-build, hibernate5-build, publish]
```
Is it enough with `publish` here, as it already depends on  the other jobs?

matrei commented on .github/workflows/gradle.yml:
```diff
@@ -135,13 +191,4 @@ jobs:
           repo: grails/gorm-mongodb
```
Can this step be removed now?

matrei commented on build.gradle:
```diff
@@ -65,12 +65,30 @@ def publishedProjects = [
+        'views-json-templates'
 ]
-subprojects {
+subprojects { Project subproject ->
```
Do we need the explicit closure parameter? Is it not already the delegate for the closure?

matrei commented on build.gradle:
```diff
@@ -65,12 +65,30 @@ def publishedProjects = [
         apply plugin: 'org.grails.grails-publish'
     }
+
+    // asset-pipeline #177 work around for parallel builds
```
Link to the issue?
https://github.com/bertramdev/asset-pipeline/issues/177

matrei commented on build.gradle:
```diff
@@ -65,12 +65,30 @@ def publishedProjects = [
+    subproject.afterEvaluate {
+        Task assetTask = subproject.tasks.findByName('assetCompile')
+        if(assetTask) {
+            assetTask.outputs.dir rootProject.layout.buildDirectory.dir("asset-serialize")
```
Single quotes for strings?

matrei commented on gradle/documentation-config.gradle:
```diff
@@ -56,10 +56,25 @@ tasks.register('data-mapping-groovydoc', Groovydoc) {
+tasks.register('copyGormImplementationDocs', Copy) {
+    dependsOn(':grails-data-hibernate5-docs:docs', ':grails-data-mongodb-docs:docs')
+
+    def hibernate5SourceDir =  project(':hibernate5-docs').layout.buildDirectory.dir('docs')
```
`grails-data-hibernate5-docs`?

matrei commented on gradle/documentation-config.gradle:
```diff
@@ -56,10 +56,25 @@ tasks.register('data-mapping-groovydoc', Groovydoc) {
+    dependsOn(':grails-data-hibernate5-docs:docs', ':grails-data-mongodb-docs:docs')
+
+    def hibernate5SourceDir =  project(':hibernate5-docs').layout.buildDirectory.dir('docs')
+    def mongodbSourceDir =  project(':mongodb-docs').layout.buildDirectory.dir('docs')
```
`grails-data-mongodb-docs`?  (also needs changes in `settings.gradle`)

matrei commented on grails-data-hibernate5/docs/build.gradle:
```diff
@@ -0,0 +1,128 @@
+            'sourcedir'     : "$project.rootDir",
+            'migrationPluginExamplesDir' : rootProject.layout.projectDirectory.dir('grails-database-migration/src/integration-test/resources').asFile.absolutePath,
+            'migrationPluginGroupId' : rootProject.findProject(':database-migration').group,
+            'migrationPluginArtifactId' : rootProject.findProject(':database-migration').name,
```
`database-migration-hibernate5`?

matrei commented on gradle/documentation-config.gradle:
```diff
@@ -56,10 +56,25 @@ tasks.register('data-mapping-groovydoc', Groovydoc) {
     }
 }
 
+tasks.register('copyGormImplementationDocs', Copy) {
```
```console
Execution failed for task ':copyGormImplementationDocs'.
> Entry api/inherit.gif is a duplicate but no duplicate handling strategy has been set.
```

matrei commented on gradle/hibernate5-test-config.gradle:
```diff
@@ -0,0 +1,53 @@
+
+        if(project.hasProperty('onlyDatastoreTests')) {
+            return false
+        }
```
Space between `if` and `(`?

matrei commented on gradle/mongodb-test-config.gradle:
```diff
@@ -0,0 +1,48 @@
+
+        if(project.hasProperty('onlyDatastoreTests')) {
+            return false
+        }
```
Space between `if` and `(`?

matrei commented on gradle/publish-config.gradle:
```diff
@@ -14,4 +19,7 @@ extensions.configure(GrailsPublishExtension) {
     ]
+    if (findProperty('pomArtifactId')) {
+        it.artifactId = findProperty('pomArtifactId')
+    }
```
`it.artifactId = findProperty('pomArtifactId') ?: name`?

matrei commented on gradle/test-config.gradle:
```diff
@@ -1,8 +1,28 @@
+
+        if(project.hasProperty('onlyHibernate5Tests')) {
+            return false
+        }
```
Space between `if` and `(`?

matrei commented on grails-data-mongodb/boot-plugin/build.gradle:
```diff
@@ -0,0 +1,87 @@
+
+ext {
+    apiDocs = true
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/libs-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on grails-data-mongodb/buildSrc/build.gradle:
Remove?

matrei commented on grails-data-mongodb/grails-plugin/build.gradle:
```diff
@@ -0,0 +1,114 @@
+ext {
+    apiDocs = true
+    pomArtifactId = 'mongodb'
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/plugins3-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on grails-data-mongodb/grails-datastore-gorm-bson/build.gradle:
```diff
@@ -0,0 +1,82 @@
+
+ext {
+    apiDocs = true
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/libs-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on grails-data-mongodb/grails-datastore-gorm-mongodb-ext/build.gradle:
```diff
@@ -0,0 +1,71 @@
+
+ext {
+    apiDocs = true
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/libs-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on grails-data-mongodb/grails-datastore-gorm-mongodb/build.gradle:
```diff
@@ -0,0 +1,136 @@
+
+ext {
+    apiDocs = true
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/libs-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on grails-data-mongodb/json-templates/build.gradle:
```diff
@@ -0,0 +1,59 @@
+    pomTitle = 'Grails MongoDB JSON Views'
+    pomDescription = 'Provides JSON Views for MongoDB to the Grails framework.'
+    pomDevelopers = ['puneetbehl': 'Puneet Behl']
+    mavenSnapshotPublishUrl = 'https://repo.grails.org/grails/plugins3-snapshots-local'
```
`snapshotPublishUrl`?

matrei commented on hibernate5/grails-datastore-gorm-hibernate/src/test/groovy/org/grails/orm/hibernate/.DS_Store:
Remove and add to `.gitignore`?

jdaugherty replied:
I can't seem to remove this file since I'm on a mac.  Is this something you can assist with?

jdaugherty replied:
I was able to get it removed, thank you.

matrei commented on settings.gradle:
```diff
@@ -61,3 +61,97 @@ project(':docs-guide-developer').projectDir = file('docs/guide-developer')
+project(":grails-datastore-gorm-mongodb-ext").projectDir = new File(settingsDir, 'grails-data-mongodb/grails-datastore-gorm-mongodb-ext')
+
+// documentation
+include 'mongodb-docs'
+project(':mongodb-docs').projectDir = new File(settingsDir, 'grails-data-mongodb/docs')
```
`grails-data-mongodb-docs`?

matrei commented on settings.gradle:
I find it difficult to navigate the project names (Gradle view).
Could we use some kind of "namespacing" to make it easier to navigate?

mongodb-examples-*
mongodb-grails-plugin
mongodb-boot-plugin
mongodb-views-json-templates
mongodb-docs
mongodb-gorm
mongodb-gorm-bson
mongodb-gorm-ext

and likewise for the others....

matrei reviewed (changes requested):
I'm not sure the TCK tests are run at the moment, and I don't think the documentation output is what it needs to be (after fixing some issues so it runs).

jdaugherty committed:
feedback:
* rename projects to have common prefixes
* spell mistakes
* build action cleanup
* update documentation
* fix styling
* serialize mongo tests because they have a shared resource (the db)
* update asset serialization (asset-pipeline#177) to be lazy

jdaugherty committed:
feedback: move up permissions where possible

jdaugherty committed:
Remove .DS_Store

jdaugherty committed:
fix: force tests to rerun in build

jdaugherty committed:
refactor: move the tck tests to main so they can be exported and imported to other projects

jdaugherty committed:
fix the tck so it always runs
* split the simple map datastore into it's own library
* split the tck up into domains / base spec / tests
* ensure the tck only runs on the simple, hibernate5, and mongo implementations

jdaugherty committed:
feedback: rename simple datastore implementation

jdaugherty commented:
@jamesfredley and I found that the tck tests weren't always running.  At least for the gorm-test / simple datastore tests, they're not running in 8.x.  You can confirm this by Changing the `SimpleMapTestSuite` and setting the test to `BuiltinUniqueConstraintWorksWithTargetProxiesConstraintsSpec`.

They all run in hibernate5 / mongodb / simple datastore.  Some of the hibernate/mongo tests had to be updated to work so I suspect they weren't all running.

While there aren't failures in mongo/hibernate, there are failures in the simple datastore.  I've left them un-ignored for now - it's 4 test classes.  We should talk about how to handle those scenarios.  

As part of "fixing" the intellij bug that was the original blocker, I've split the tck up into 3 packages.  This lets intellij behave - it was locking up in the Grails plugin when the tck project was inspected.  I suspect it's because we were replacing class files and there were duplicates in various modules.  

I've also split up the simple data store away from the tests, and the test library no longer is published.  We can decide if we want to publish our tests, but I don't think any of our projects do that.  The tck is obviously special and is still published.

jdaugherty committed:
ignore tests that were failing in prior releases

jamesfredley committed:
Adjust @PendingFeatureIf conditions on FirstAndLastMethodSpec

jamesfredley committed:
Merge branch 'newMongoMerge' of https://github.com/jdaugherty/grails-data-mapping into newMongoMerge

jdaugherty committed:
tests: do not not ignore tests in mongo that pass

jdaugherty commented:
we added pending feature for the failing tests in 8.x to make a clean test run

matrei commented on docs/guide-developer/src/main/docs/stepByStep.adoc:
```diff
@@ -32,7 +32,7 @@ dependencies {
             project(':grails-datastore-gorm-support')
             
-    testImplementation project(':grails-datastore-gorm-tck')
+    testImplementation project(':grails-datastore-gorm-tck').testFixtures
```
Is this correct?

matrei commented on docs/guide-developer/src/main/docs/testing.adoc:
```diff
@@ -2,7 +2,7 @@ The `grails-datastore-gorm-tck` project provides several hundred tests to guaran
 [source,groovy]
 ----
-testCompile project(':grails-datastore-gorm-tck')
+testCompile project(':grails-datastore-gorm-tck').testFixtures
```
Is this correct?

matrei commented on gradle/mongodb-test-config.gradle:
```diff
@@ -8,27 +8,26 @@ tasks.named('compileTestGroovy', GroovyCompile) {
     useJUnitPlatform()
+    maxParallelForks = 1
     if (isCiBuild) {
-        maxParallelForks = 1
         forkEvery = 10
     } else {
-        maxParallelForks = 1
         forkEvery = 30
     }
```
Does it make sense to set `forkEvery` when `maxParallelForks = 1`?

jdaugherty replied:
I assume it's to reduce memory usage - these were the settings before so I'm inclined to leave them.

matrei replied:
I might misunderstand how this works, but if it is not allowed to fork, what is the use to set `forkEvery`?

matrei commented on gradle/tck-config.gradle:
```diff
@@ -1,53 +1,33 @@
 dependencies {
+    if(!project.hasProperty('includeBaseTckClass') || project.findProperty('includeBaseTckClass')) {
```
Space between `if` and `(`?
Is this condition correct? Will it ever not be `true`?

jdaugherty replied:
I did this for completeness but right now the property is either set to true or is missing.

matrei commented on gradle/tck-config.gradle:
```diff
@@ -1,53 +1,33 @@
-// project into this project's test classes dir so Gradle can find the test
-// classes and run them. See grails.gorm.tests.GormDatastoreSpec for on the TCK.
+// TODO: gradle will happily put the jar file on the classpath, but junit won't find the tests.  Dynamic test discovery also won't find them.
+tasks.register("extractTckJar", Copy) {
```
Single quotes for strings?

matrei commented on gradle/tck-config.gradle:
```diff
@@ -1,53 +1,33 @@
+tasks.register("extractTckJar", Copy) {
+    dependsOn(configurations.testRuntimeClasspath)
+
+    Provider<Directory> extractTarget = layout.buildDirectory.dir("extracted-tck-classes")
```
Single quotes for strings?

matrei commented on gradle/tck-config.gradle:
```diff
@@ -1,53 +1,33 @@
 
-tasks.named('test') {
+tasks.withType(Test).configureEach {
+    dependsOn("extractTckJar")
```
Single quotes for strings?

matrei commented on gradle/tck-config.gradle:
```diff
@@ -1,53 +1,33 @@
-                file("src/test/groovy/${relativePath}.groovy").exists()
-            }
-        }
+        testClassesDirs += files(layout.buildDirectory.dir("extracted-tck-classes"))
```
Single quotes for strings?

matrei commented on grails-data-graphql/examples/spring-boot-app/build.gradle:
```diff
@@ -1,12 +1,10 @@
 	}
 	dependencies {
-		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+		implementation platform("org.grails:grails-bom:$grailsVersion")
```
Is implementation correct here?

matrei commented on grails-data-hibernate5/gorm/src/test/groovy/grails/gorm/tests/GormDatastoreSpec.groovy:
```diff
@@ -0,0 +1,106 @@
+ */
+abstract class GormDatastoreSpec extends Specification {
+
+    static final CURRENT_TEST_NAME = "current.gorm.test"
```
Single quotes for strings?

matrei commented on grails-data-hibernate5/gorm/src/test/groovy/grails/gorm/tests/GormDatastoreSpec.groovy:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    def setup() {
+        System.out.println("Using hibernate5 datastore class")
```
Do we need `System.out`?
Single quotes for strings?

jdaugherty replied:
I added this to make it easier to confirm it's pulling the right base class.  I really don't like the way the TCK was implemented because it's causing duplicate class names that conflict.  By separating them into a separate base package, I was able to "swap" them in / out so IntelliJ behaves, but longer term I think we should switch this to a generator approach that copies the class files.  That's why I repackaged.  

This change is far too big for this code review though, so it can be done in the future.  at least it works now ...

matrei commented on grails-datastore-gorm-tck-base/src/main/groovy/grails/gorm/tests/GormDatastoreSpec.groovy:
```diff
@@ -37,6 +62,7 @@ abstract class GormDatastoreSpec extends Specification {
     }
 
     def setup() {
+        System.out.println("Using base datastore class")
```
Do we need `System.out`?
Single quotes for strings?

matrei commented on grails-datastore-gorm-tck/src/main/groovy/grails/gorm/tck/BuiltinUniqueConstraintWorksWithTargetProxiesConstraintsSpec.groovy:
```diff
@@ -1,10 +1,11 @@
 
 class BuiltinUniqueConstraintWorksWithTargetProxiesConstraintsSpec extends GormDatastoreSpec {
-
+    @PendingFeatureIf({ !Boolean.getBoolean("hibernate5.gorm.suite") && !Boolean.getBoolean("hibernate6.gorm.suite") && !Boolean.getBoolean("mongodb.gorm.suite")})
```
Single quotes for strings?

matrei commented on grails-datastore-gorm-tck/src/main/groovy/grails/gorm/tck/BuiltinUniqueConstraintWorksWithTargetProxiesConstraintsSpec.groovy:
```diff
@@ -25,6 +26,7 @@ class BuiltinUniqueConstraintWorksWithTargetProxiesConstraintsSpec extends GormD
         ContactDetails.deleteAll(contactDetails1)
     }
 
+    @PendingFeatureIf({ !Boolean.getBoolean("hibernate5.gorm.suite") && !Boolean.getBoolean("hibernate6.gorm.suite") && !Boolean.getBoolean("mongodb.gorm.suite")})
```
Single quotes for strings?

matrei commented on grails-datastore-gorm-tck/src/main/groovy/grails/gorm/tck/DirtyCheckingSpec.groovy:
```diff
@@ -151,7 +152,7 @@ class DirtyCheckingSpec extends GormDatastoreSpec {
     }
 
-    @IgnoreIf({ Boolean.getBoolean("hibernate5.gorm.suite") || Boolean.getBoolean("hibernate6.gorm.suite")}) // because one-to-one association loads eagerly in the Hibernate
+    @IgnoreIf({ !Boolean.getBoolean("mongodb.gorm.suite")})
```
Single quotes for strings?

matrei commented on grails-datastore-gorm-tck/src/main/groovy/grails/gorm/tck/FindByMethodSpec.groovy:
```diff
@@ -141,7 +145,7 @@ class FindByMethodSpec extends GormDatastoreSpec {
 
         when:
-            Book.newInstance(author: 'Jeff', title: 'Fly Fishing For Everyone', published: false).save()
+        Book.newInstance(author: 'Jeff', title: 'Fly Fishing For Everyone', published: false).save()
```
Spacing?

matrei commented on grails-datastore-gorm-tck/src/main/groovy/grails/gorm/tck/ValidationSpec.groovy:
```diff
@@ -127,41 +178,74 @@ class ValidationSpec extends GormDatastoreSpec {
     }
 
+    @IgnoreIf({
+        Boolean.getBoolean("neo4j.gorm.suite") || // neo4j requires a transaction present for inserts
```
Single quotes for strings?

matrei commented on grails-datastore-gorm-test/src/test/groovy/org/grails/datastore/gorm/NotLikeSpec.groovy:
```diff
@@ -1,13 +1,19 @@
 class NotLikeSpec extends GormDatastoreSpec {
 
+    def setup() {
+        System.out.println("********************************************************")
+        System.out.println(System.getProperty("java.class.path"))
+        System.out.println("********************************************************")
+    }
```
Do we need `System.out`?
Single quotes for strings?

jdaugherty replied:
Woops, this was when I was debugging.  Removed.

matrei commented on settings.gradle:
Thank you for namespacing the project names!
Can we also have `hibernate5-examples-*`?

matrei reviewed (approved):
This was a beast! Good work!

jdaugherty committed:
feedback: styling & removing debug
